pkg/tasks/grafana.go

```diff
@@ -42,99 +39,14 @@ func NewGrafanaTask(client *client.Client, factory *manifests.Factory, config *m
 }
 
 func (t *GrafanaTask) Run(ctx context.Context) error {
-	if t.config.ClusterMonitoringConfiguration.GrafanaConfig.IsEnabled() {
-		return t.create(ctx)
+	if err := t.create(ctx); err != nil {
+		return err
 	}
 
-	return t.destroy(ctx)
+	return t.cleanup(ctx)
 }
 
 func (t *GrafanaTask) create(ctx context.Context) error {
-	cr, err := t.factory.GrafanaClusterRole()
-	if err != nil {
-		return errors.Wrap(err, "initializing Grafana ClusterRole failed")
-	}
-
-	err = t.client.CreateOrUpdateClusterRole(ctx, cr)
-	if err != nil {
-		return errors.Wrap(err, "reconciling Grafana ClusterRole failed")
-	}
-
-	crb, err := t.factory.GrafanaClusterRoleBinding()
-	if err != nil {
-		return errors.Wrap(err, "initializing Grafana ClusterRoleBinding failed")
-	}
-
-	err = t.client.CreateOrUpdateClusterRoleBinding(ctx, crb)
-	if err != nil {
-		return errors.Wrap(err, "reconciling Grafana ClusterRoleBinding failed")
-	}
-
-	r, err := t.factory.GrafanaRoute()
-	if err != nil {
-		return errors.Wrap(err, "initializing Grafana Route failed")
-	}
-
-	err = t.client.CreateRouteIfNotExists(ctx, r)
-	if err != nil {
-		return errors.Wrap(err, "creating Grafana Route failed")
-	}
-
-	_, err = t.client.WaitForRouteReady(ctx, r)
-	if err != nil {
-		return errors.Wrap(err, "waiting for Grafana Route to become ready failed")
-	}
-
-	rs, err := t.factory.GrafanaRBACProxyMetricSecret()
-	if err != nil {
-		return errors.Wrap(err, "initializing Grafana RBAC proxy metric Secret failed")
-	}
-
-	err = t.client.CreateIfNotExistSecret(ctx, rs)
-	if err != nil {
-		return errors.Wrap(err, "creating Grafana RBAC proxy metric Secret failed")
-	}
-
-	ps, err := t.factory.GrafanaProxySecret()
-	if err != nil {
-		return errors.Wrap(err, "initializing Grafana proxy Secret failed")
-	}
-
-	err = t.client.CreateIfNotExistSecret(ctx, ps)
-	if err != nil {
-		return errors.Wrap(err, "creating Grafana proxy Secret failed")
-	}
-
-	smc, err := t.factory.GrafanaConfig()
-	if err != nil {
-		return errors.Wrap(err, "initializing Grafana Config Secret failed")
-	}
-
-	err = t.client.CreateOrUpdateSecret(ctx, smc)
-	if err != nil {
-		return errors.Wrap(err, "reconciling Grafana Config Secret failed")
-	}
-
-	sds, err := t.factory.GrafanaDatasources()
-	if err != nil {
-		return errors.Wrap(err, "initializing Grafana Datasources Secret failed")
-	}
-
-	err = t.client.CreateIfNotExistSecret(ctx, sds)
-	if err != nil {
-		return errors.Wrap(err, "reconciling Grafana Datasources Secret failed")
-	}
-
-	err = t.client.DeleteSecret(ctx, &v1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      grafanaLegacyDataSourceSecretName,
-			Namespace: sds.GetNamespace(),
-		},
-	})
-	if err != nil {
-		return errors.Wrap(err, "deleting legacy Grafana Datasources secret failed")
-	}
-
 	cmdds, err := t.factory.GrafanaDashboardDefinitions()
```

Comment:
I don't think we even need this. https://github.com/openshift/cluster-monitoring-operator/blob/master/manifests/0000_90_cluster-monitoring-operator_01-dashboards.yaml would serve the console dashboard needs which is be created here https://github.com/openshift/cluster-monitoring-operator/blob/master/hack/build-jsonnet.sh#L37

Reply:
Done